packages/next/src/server/stream-utils/node-web-streams-helper.ts

```diff
@@ -72,15 +69,14 @@ export function chainStreams<T>(
 }
 
 export function streamFromArray(strings: string[]): ReadableStream<Uint8Array> {
-  // Note: we use a TransformStream here instead of instantiating a ReadableStream
-  // because the built-in ReadableStream polyfill runs strings through TextEncoder.
-  const { readable, writable } = new TransformStream()
-
-  const writer = writable.getWriter()
-  strings.forEach((str) => writer.write(encodeText(str)))
-  writer.close()
-
-  return readable
+  return new ReadableStream({
+    start(controller) {
+      for (const str of strings) {
+        controller.enqueue(encodeText(str))
+      }
+      controller.close()
+    },
+  })
```

Comment:
Suspect that this TransformStream trick is no longer needed.

Reply:
Sounds like a todo to confirm